apps/cache-manager/src/app.ts

```diff
@@ -0,0 +1,42 @@
+import express from 'express'
+
+import {CONNECTION_STRING, TABLE_NAME} from './globals'
+import {AzureTableStorage, cacheExpired, getComponent, insertNewItem} from './utils'
+
+const app = express()
+app.use(express.json())
+
+app.post('/:port/prerender', async (req, res) => {
+  try {
+    let component
+
+    const remoteName = req.get('remote-name') || 'batman'
+
+    const client = await AzureTableStorage.connectTableClient(CONNECTION_STRING, TABLE_NAME)
+
+    const base64Body = Buffer.from(JSON.stringify(req.body)).toString('base64')
+
+    const tableRes: any = await AzureTableStorage.getTableItem(client, remoteName, base64Body)
+
+    if (tableRes === undefined) {
+      const response = await getComponent(req.body, req.params.port)
+      component = response.data
+      await insertNewItem(remoteName, base64Body, response.data, client)
+    } else if (cacheExpired(tableRes.expiryDate)) {
+      await AzureTableStorage.deleteTableItem(client, tableRes.partitionKey, tableRes.rowKey)
+      const response = await getComponent(req.body, req.params.port)
+      await insertNewItem(remoteName, base64Body, response.data, client)
+      component = response.data
+    } else {
+      component = tableRes.value
+    }
+
+    res.status(200).send(component)
+  } catch (err) {
+    console.log('error try catch')
```

Comment:
Should we fallback to getComponent?

Reply:
hmm, the condition is if (tableRes === undefined), so in some ways the first part is a fallback... Perhaps we flip it round so the condition is if (tableRes), then the fallback will be the generic getcomponent?

Reply:
If the Table Storage service is unavailable does it return undefined or throw an error trying to connect?

Reply:
oh right, sorry misunderstood you there, hmm, we've not handled failure well yet i don't think, can look at doing so now, but had in my head that this would be a separate ticket, although having thought about it it is kinda required for MVP

Reply:
Just think if it all still works without setting up the azurite service its probably a good thing, or if the service connection goes down the site doesn't fall over. We can look at enhancing it later.